qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp

```diff
@@ -421,24 +421,41 @@ void IndirectFitAnalysisTab::updateParameterValues(
 }
 
 void IndirectFitAnalysisTab::updateFitBrowserParameterValues() {
-  if (m_fittingAlgorithm) {
-    MantidQt::API::SignalBlocker blocker(m_fitPropertyBrowser);
-    if (m_fittingModel->getFittingMode() == FittingMode::SEQUENTIAL) {
-      auto const paramWsName =
-          m_fittingAlgorithm->getPropertyValue("OutputParameterWorkspace");
-      auto paramWs =
-          AnalysisDataService::Instance().retrieveWS<ITableWorkspace>(
-              paramWsName);
-      auto rowCount = static_cast<int>(paramWs->rowCount());
-      if (TableRowIndex{rowCount} == m_fittingModel->getNumberOfDomains())
-        m_fitPropertyBrowser->updateMultiDatasetParameters(*paramWs);
-    } else {
-      IFunction_sptr fun = m_fittingAlgorithm->getProperty("Function");
-      if (fun->getNumberDomains() > 1)
-        m_fitPropertyBrowser->updateMultiDatasetParameters(*fun);
-      else
-        m_fitPropertyBrowser->updateParameters(*fun);
+  IFunction_sptr fun = m_fittingModel->getFittingFunction();
+  if (fun->getNumberDomains() > 1)
+    m_fitPropertyBrowser->updateMultiDatasetParameters(*fun);
+  else
+    m_fitPropertyBrowser->updateParameters(*fun);
+}
+
+void IndirectFitAnalysisTab::updateFitBrowserParameterValuesFromAlg() {
+  try {
```

Comment:
Can we reduce the scope of the try catch block again?

Reply:
I think in this case the exceptions can be thrown from several places through the code block. Whilst I agree that ideally it would be smaller doing so feels a bit more dangerous in terms of panhandled exceptions than I want to do now.